src/hooks/usePantry.ts

```diff
@@ -181,9 +181,52 @@ function entry({ project }: { project: string }): Entry {
     const versions = dir.join("versions.txt")
     return { dir, yml, versions }
   }
-  throw new TeaError('not-found: pantry: package.yml', {project})
-}
+  //check for fuzzy string
+  const closestPkg = getClosestPackageSuggestion(project, prefix);
+  closestPkg.then(function(data){
+    console.log("did you mean", {data}, "?")
+  });
+  throw new TeaError('not-found: pantry: package.yml', {project}, )
 
+}
+async function getClosestPackageSuggestion(orgPkg:string, projectPath:Path): Promise<string>{
+  let closestPkg = '';
+  let minDistance = Infinity;
+  const pkgList = [];
+  for await (const data of projectPath.ls()){
+    const {name} = data[1];
+    pkgList.push(name);
+  }
+  pkgList.forEach(pkgName=>{
+    const number = levenshteinDistance(pkgName, orgPkg);
+    if(number<minDistance){
+      minDistance = number;
+      closestPkg = pkgName;
+    }
+  })
+  return closestPkg;
```

Comment:
Instead of always returning `closestPkg`, Can we use threshold here?
Something like if we don't find a package related to the ones we support.

like `random12323`

Reply:
Sure, The algorithm works in such a way, if you need to convert string 1 to string 2 then it tells us the minimum number of operations required using only three operations.
1. add character
2. modify character
3. remove character

So if the value of operations is greater than 2 or 3 we can avoid showing the suggestion. It won't be a typo.

Reply:
oh Ya That could work